src/csharp/Grpc.Core/Internal/DefaultDeserializationContext.cs

```diff
@@ -46,10 +46,12 @@ public DefaultDeserializationContext()
 
         public override byte[] PayloadAsNewBuffer()
         {
+            if (payloadLength == 0) return EmptyBlob;
```

Comment:
In this case we should also update the xmldoc of the `PayloadAsNewBuffer` method:
https://github.com/grpc/grpc/blob/72b95e18fafbd3e2c66b0480f9879478da95ab65/src/csharp/Grpc.Core.Api/DeserializationContext.cs#L34